crates/net/network/src/session/mod.rs

```diff
@@ -552,12 +554,72 @@ impl SessionManager {
                 self.remove_pending_session(&session_id);
                 match direction {
                     Direction::Incoming => {
+                        match error.as_ref().and_then(|e| e.as_eth_stream()) {
```

Comment:
this is just a match over an error, right?

so I'd like to extract this to a function instead

Reply:
agreed, smthg like `SessionManager::update_disconnect_incoming_metrics`